src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFrameworkCore.psm1

```diff
@@ -165,847 +97,755 @@ Register-TabExpansion Drop-Database @{
-            Invoke-Process -Executable $exe -Arguments $arguments -RedirectByPrefix -ErrorAction SilentlyContinue -ErrorVariable invokeErrors -JsonOutput | ConvertFrom-Json
-        } else {
-            # don't capture output. Output lines flow through
-            Invoke-Process -Executable $exe -Arguments $arguments -RedirectByPrefix -ErrorAction SilentlyContinue -ErrorVariable invokeErrors
-        }
+    if ($json)
+    {
+        $params += '--json'
 
-    } finally {
-        if ($exeCopied) {
-            Write-Debug "Cleaning up '$exe' and '$configFile'"
-            Remove-Item $exe -ErrorAction SilentlyContinue | Out-Null
-            if ($configFile) {
-                Remove-Item $configFile -ErrorAction SilentlyContinue | Out-Null
-            }
-        }
+        Invoke-Process -Executable $efPath -Arguments $params -RedirectByPrefix -ErrorAction SilentlyContinue -ErrorVariable invokeErrors -JsonOutput | ConvertFrom-Json
+    }
+    else
+    {
+        Invoke-Process -Executable $efPath -Arguments $params -RedirectByPrefix -ErrorAction SilentlyContinue -ErrorVariable invokeErrors
     }
 
-    if ($invokeErrors) {
-        $combined = ($invokeErrors | `
-            ? { !($_.Exception.Message -like '*non-zero exit code')} | `
-            % { $_.Exception.Message }) -join "`n"
-        if (!$combined) {
-            $lastError = $invokeErrors | Select-Object -Last 1
-            if (!$lastError.Exception.Message) {
+    if ($invokeErrors)
+    {
+        $combined = ($invokeErrors |
+            ?{ $_.Exception.Message -notLike '*non-zero exit code' } |
+            %{ $_.Exception.Message }) -join "`n"
+        if (!$combined)
+        {
+            $lastError = $invokeErrors | select -Last 1
+            if (!$lastError.Exception.Message)
+            {
                 throw 'Operation failed with unspecified error'
             }
 
             throw $lastError.Exception.Message
         }
+
         throw $combined
     }
 }
 
+function IsWeb($project)
+{
+    $types = GetProjectTypes $project
+
+    return $types -contains '{349C5851-65DF-11DA-9384-00065B846F21}'
+}
+
+function IsUWP($project)
+{
+    $types = GetProjectTypes $project
+
+    return $types -contains '{A5A43C5B-DE2A-4C0C-9213-0A381AF9435A}'
+}
+
+function GetProjectTypes($project)
+{
+    $solution = Get-VSService 'Microsoft.VisualStudio.Shell.Interop.SVsSolution' 'Microsoft.VisualStudio.Shell.Interop.IVsSolution'
+    $hierarchy = $null
+    $hr = $solution.GetProjectOfUniqueName($project.UniqueName, [ref] $hierarchy)
+    [Runtime.InteropServices.Marshal]::ThrowExceptionForHR($hr)
+
+    $aggregatableProject = Get-Interface $hierarchy 'Microsoft.VisualStudio.Shell.Interop.IVsAggregatableProject'
+
+    $projectTypeGuidsString = $null
+    $hr = $aggregatableProject.GetAggregateProjectTypeGuids([ref] $projectTypeGuidsString)
+    [Runtime.InteropServices.Marshal]::ThrowExceptionForHR($hr)
+
+    return $projectTypeGuidsString.Split(';')
+}
```

Comment:
Not being Jedi enough to write this function was one reason the EF6 cmdlets were written in C#. 😉